packages/core/src/app/customer/Customer.tsx

```diff
@@ -405,7 +405,9 @@ class Customer extends Component<CustomerProps & WithCheckoutCustomerProps & Ana
 
             const customer = data.getCustomer();
 
-            if (customer && customer.shouldEncourageSignIn && customer.isGuest && !customer.isStripeLinkAuthenticated) {
+            const paymentProviderCustomer = data.getPaymentProviderCustomer();
+
+            if (customer && customer.shouldEncourageSignIn && customer.isGuest && !paymentProviderCustomer?.authenticationState) {
```

Comment:
This line might throw an error after you pull last changes from master. May I ask you to pull and check?

Reply:
I changed it to `stripeLinkAuthenticationState` as it was needed. After pulling newest changes everything is working fine.